database.c

```diff
@@ -307,17 +307,17 @@ static const char *current_(const char *profile, const char *key)
   return res;
 }
 
-/* ========================================================================= *
- * FIXME: horrible $HOME/.profiled access hacks
- * ========================================================================= */
-
 static const char *datadir(void)
 {
   static char path[256] = "";
 
   if( *path == 0 )
   {
-    snprintf(path, sizeof path, "%s/.profiled", getenv("HOME") ?: "/tmp");
+    if (snprintf(path, sizeof(path), "%s/.config/profiled", getenv("HOME")) < 0)
+    {
+      log_err("Path %s/.config/profiled is too long. Max size %lu\n", getenv("HOME"), sizeof(path));
+      snprintf(path, sizeof(path), "/tmp");
```

Comment:
This goes all wrong now...

Old code: protected against side effects of NULL $HOME and dontcare if $HOME is bonkers long (under assumption that it will be checked later on or something).

New code: removes NULL protection, adds incorrect "path too long" check (snprintf returns number of chars it would have written if there were space, negative values are something else) and superfluous parentheses for sizeof...

Reply:
Please don't hardcode $HOME/.config but use $XDG_CONFIG_HOME